lib/ex_activecampaign/api_v1.ex

```diff
@@ -0,0 +1,236 @@
+
+      iex> ExActivecampaign.ApiV1.content_type([{"Content-Type", "application/xml; charset=utf-8"}])
+      "application/x-www-form-urlencoded"
+
+      iex> ExActivecampaign.ApiV1.content_type([{"Server", "GitHub.com"}, {"Content-Type", "application/xml; charset=utf-8"}])
+      "application/x-www-form-urlencoded"
+  """
+  def content_type({status, body, headers}), do: {status, body, content_type(headers)}
+  def content_type([]), do: "application/x-www-form-urlencoded"
+  def content_type([{"Content-Type", _val} | _]), do: "application/x-www-form-urlencoded"
+  def content_type([_ | tail]), do: tail |> content_type
+
+  @doc """
+  Encode the body to pass along to the server
+
+  ## Examples
+
+      iex> ExActivecampaign.ApiV1.encode(%{a: 1}, "application/json")
+      {:error, "Unsupported Content-Type for API V1 requests"}
+
+      iex> ExActivecampaign.ApiV1.encode(nil, "application/json")
+      {:error, "Unsupported Content-Type for API V1 requests"}
+
+      iex> ExActivecampaign.ApiV1.encode("<xml/>", "application/xml")
+      {:error, "Unsupported Content-Type for API V1 requests"}
+
+      iex> ExActivecampaign.ApiV1.encode(%{a: "o ne"}, "application/x-www-form-urlencoded")
+      "a=o+ne"
+
+      iex> ExActivecampaign.ApiV1.encode("goop", "application/stuff")
+      {:error, "Unsupported Content-Type for API V1 requests"}
+  """
+  def encode(data, "application/x-www-form-urlencoded"), do: URI.encode_query(data)
+  def encode(_, _), do: {:error, "Unsupported Content-Type for API V1 requests"}
+
+  @doc """
+  Decode the response body
+
+  ## Examples
+
+      iex> ExActivecampaign.ApiV1.decode({:ok, "{\\"a\\": 1}", "application/json"})
+      {:ok, %{"a" => 1}}
+
+      iex> ExActivecampaign.ApiV1.decode({500, "", "application/json"})
+      {500, ""}
+
+      iex> ExActivecampaign.ApiV1.decode({:error, "{\\"a\\": 1}", "application/json"})
+      {:error, %{"a" => 1}}
+
+      iex> ExActivecampaign.ApiV1.decode({:ok, "{goop}", "application/json"})
+      {:error, "{goop}"}
+
+      iex> ExActivecampaign.ApiV1.decode({:error, "{goop}", "application/json"})
+      {:error, "{goop}"}
+
+      iex> ExActivecampaign.ApiV1.decode({:error, :nxdomain, "application/dontcare"})
+      {:error, :nxdomain}
+  """
+  def decode({status, "", _}), do: {status, ""}
+
+  def decode({_status, body, "application/xml"}) do
+    decoded_body = body |> :binary.bin_to_list() |> :xmerl_scan.string()
+
+    case decoded_body do
+      {:exit, _e} -> {:error, body}
+      _ -> {:ok, decoded_body}
+    end
+  end
+
+  def decode({status, body, _}) when is_binary(body) do
+    body
+    |> Poison.decode()
+    |> case do
+      {:ok, parsed} -> {status, parsed}
+      _ -> {:error, body}
+    end
+  end
+
+  def decode({status, body, _}), do: {status, body}
+
+  @doc """
+  Clean the URL, if there is a port, but nothing after, then ensure there's a ending '/' otherwise you will encounter
+  something like hackney_url.erl:204: :hackney_url.parse_netloc/2
```

Comment:
Minor niggle but could this be changed to say the trailing '/' is after the port number? Otherwise suggests this will return something like http://localhost:8081/v1/profile/